File: 2.POO/Resto/Proyecto/Java/Financiera.java
import java.io.*;
import java.util.Scanner;
public class Financiera {
    public static void main(String[] args) {
        Scanner mex= new Scanner(System.in);
        int MAX_ELEMENS =100;
        Devolucion D1[] = new Devolucion[MAX_ELEMENS];
        for(int i=0; i<MAX_ELEMENS;i++){
            D1[i]= new Devolucion();
        }

        Prestamo P1[] = new Prestamo[MAX_ELEMENS];
        for(int i=0; i<MAX_ELEMENS; i++){
            P1[i]= new Prestamo();
        }

        Cambio C1 = new Cambio();
        int i=0, respuesta, D=0, P=0, a;
        String fecha, clave[][]= new String [50][3];
        double acumulador=0,contador=0,ganancia=0;
        
        System.out.println("Bienvenido a la financiera JAMD. \n");
        System.out.println("Ingresa la fecha: ");
        fecha=mex.nextLine();

        do {
            if (i<MAX_ELEMENS){
                System.out.println("\n");
                System.out.println("Al programa le quedan " + MAX_ELEMENS + " intentos");
                System.out.println("\n");
                System.out.println("Elige una opcion \n");
                System.out.println("1) Ingresar nuevo prestamo.");
                System.out.println("2) Ingrsar un abono.");
                System.out.println("3) Total recaudado de abonos.");
                System.out.println("4) Total del dinero prestado.");
                System.out.println("5) Total de ganancia esperada.");
                System.out.println("6) Calcular cambio.");
                System.out.println("7) Ver historia.");
                System.out.println("8) Cerrar programa.");
                System.out.println("Opcion: ");
                a=mex.nextInt();
                mex.nextLine();
 
            

                switch(a){

                    case 1:
                    try {
                        System.out.println("\n");
                        System.out.println("Ingresa la clave del cliente: ");
                        clave[i][0]= mex.nextLine();
                        clave[i][0] += ".txt";

                        try {
                            FileWriter fw = new FileWriter(clave[i][0], true);
                            fw.close();
                        } catch (Exception e) {
                            e.printStackTrace();
                        }
                        System.out.println("\n");
                        System.out.println("Ingresa el nombre: ");
                        P1[P].setNombre(mex.nextLine()); 
                        System.out.println("Ingresa el apellidos: ");
                        P1[P].setApellido(mex.nextLine());
                        System.out.println("Ingresa la cantidad a prestar: ");
                        P1[P].setCantidadPrestada(mex.nextDouble());
                        System.out.println("Ingresa el procentaje de interes: ");
                        P1[P].setCantidadDeInteres(mex.nextDouble());
                        System.out.println("Ingresa la cantidad de semanas a pagar: ");
                        P1[P].setCantidadSemanas(mex.nextDouble());
                        
                        
                    } catch (Exception e) {
                        System.out.println(e.getMessage());
                    }
                    acumulador+= P1[P].getCantidadPrestada();
                    ganancia+= P1[P].Interes(); 
                    P++;
                    break;

                    case 2:
                    try{
                        System.out.println("\n");
                        System.out.println("Ingresa la clave del cliente: ");
                        clave[i][1]= mex.nextLine();
                        clave[i][1] += ".txt";

                        try {
                            FileWriter fw = new FileWriter(clave[i][1], true);
                            fw.close();
                        } catch (Exception e) {
                            e.printStackTrace();
                        }
                        System.out.println("\n");
                        System.out.println("Ingresa el nombre: ");
                        D1[D].setNombre(mex.nextLine()); 
                        System.out.println("Ingresa el apellidos: ");
                        D1[D].setApellido(mex.nextLine());
                        System.out.println("Ingresa la cantidad prestada: ");
                        D1[D].setCantidadPrestada(mex.nextDouble());
                        System.out.println("Ingresa la cantidad devuelta: ");
                        D1[D].setCantidadDevuelta(mex.nextDouble());
                        System.out.println("Ingresa las semanas a pagar: ");
                        D1[D].setSemanasAPagar(mex.nextDouble());
                        System.out.println("Ingresa la semana pagada: ");
                        D1[D].setSemanaPagada(mex.nextDouble());
                    } catch (Exception e){
                        System.out.println(e.getMessage());
                    }
                    contador+= D1[D].getCantidadDevuelta();
                    D++;
                    break;

                    case 3:
                    System.out.println("\n");
                    System.out.println("Fecha: " + fecha);
                    System.out.println("Total de abonos: $" + contador);
                    break;

                    case 4:
                    System.out.println("\n");
                    System.out.println("Fecha: " + fecha);
                    System.out.println("Total de dinero prestado: $" + acumulador);
                    break;

                    case 5:
                    System.out.println("\n");
                    System.out.println("Fecha: " + fecha);
                    System.out.println("Ganancia esperada: $" + ganancia);
                    break;

                    case 6:
                    System.out.println("Ingresa la cantidad pagada: ");
                    C1.setPago(mex.nextDouble());
                    System.out.println("Ingresa la cantidad a pagar: ");
                    C1.setCantidad(mex.nextDouble());
                    System.out.println("\n");
                    System.out.println("Cambio: $" + C1.Buelto());

                    break;

                    case 7:
                    System.out.println("\n");
                    System.out.println("Ingresa la clave del cliente: ");
                        clave[i][2]= mex.nextLine();
                        clave[i][2] += ".txt";

                        try {
                            FileWriter fw = new FileWriter(clave[i][2], true);
                            fw.close();
                        } catch (Exception e) {
                            e.printStackTrace();
                        }

                    BufferedReader bfr;
                    try {
                        bfr = new BufferedReader(new FileReader(clave[i][2]));
                        System.out.println("Historial del cliente: ");
                        String texto = bfr.readLine();
                        while (texto != null){
                            System.out.println(texto);
                            texto=bfr.readLine();
                        }
                    } catch (Exception e) {
                        e.printStackTrace();
                    }  
                    break;

                    case 8:
                    System.out.println("Programa finalizado.");
                    respuesta = 0;
                    break;
                }
                i++;
                MAX_ELEMENS--;
                System.out.println("\n");
                System.out.println("Presione 1 se desea repetir y 0 se desea finalizar: ");
                respuesta= mex.nextInt();
                mex.nextLine();
            }else {respuesta = 0;}

           

        }while(respuesta !=0);

        try {
            int iii=0;
            
            for(int ii=0; ii<i; ii++){
                if(P1[ii].getCantidadPrestada() > 0 && P1[ii].getCantidadDeInteres() > 0 && P1[ii].getCantidadSemanas() > 0){
                    FileWriter fw = new FileWriter(clave[iii][0], true);
                    fw.write(" \n");   
                    fw.write("Prestamos: --------------------------------------------------\n");
                    fw.write("Fecha: " + fecha + "\n");
                    fw.write( P1[ii].ToString());
                    fw.write("---------------------------------------------------------------------------\n");
                    fw.close();
                    iii++;
                    
                }
            }
 
            for(int ii=0; ii<i; ii++) {
                if (D1[ii].getCantidadDevuelta() > 0 && D1[ii].getCantidadPrestada() > 0 && D1[ii].getSemanaPagada() > 0 
                && D1[ii].getSemanasAPagar() > 0){ 
                    FileWriter fw = new FileWriter(clave[iii][1], true);
                    fw.write(" \n");               
                    fw.write("Devoluciones: --------------------------------------------------------------\n");
                    fw.write("Fecha: " + fecha + "\n");
                    fw.write( D1[ii].ToString());
                    fw.write("---------------------------------------------------------------------------\n");
                    fw.close();
                    iii++;
    
                }
            }

            
            System.out.println("Se a almacenado los datos.\n");
        } catch (Exception e) {
            e.printStackTrace();
        }
               
        System.out.println("---------------------------------------------------------------------------");
        System.out.println("Creador: Jesus A. Mex Damian 24/05/2023.");

        mex.close();
    }
}

File: 2.POO/Resto/Proyecto/C++/Financiera.cpp
#include <iostream>
#include <fstream>
#include <cstdlib>
using namespace std;
#include "Devolucion.h"
#include "Prestamo.h"
#include "Seguridad.h"
#include "Cambio.h"
 


int main() {
    fstream archivo;
    system("color 02");
    cout << "\n \n";
    int MAX_ELEMENTS= 100;
    Devolucion D1[MAX_ELEMENTS];
    int D= 0;
    Prestamo P1[MAX_ELEMENTS];
    int P= 0;
    Seguridad S1;
    Cambio PA1;
    string palabras,fecha, clave[1000][3], id;
    double cantidades, contador = 0, acumulador = 0, ganancia = 0;
    int a, respuesta;
    int i= 0,c=0;

    
        

    
        cout << "\n \n";
        cout << "Bienvenido a la financiera JAMD. \n";
        cout << "Ingresa la fecha: ";
        getline(cin,fecha);
        

        do{
            cout << "\n";
            cout << "Al programa le quedan " << MAX_ELEMENTS << " intentos";
            cout << "\n";
            if (i<MAX_ELEMENTS){
                
                cout << "\n";
                cout << "Elige una opcion. \n";
                cout << "\n";
                cout << "1) Agregar prestamo. \n";
                cout << "2) Agregar devolucion. \n";
                cout << "3) Total recaudado de abonos. \n";
                cout << "4) Total del dinero prestado. \n";
                cout << "5) Total de ganancia esperada. \n";
                cout << "6) Calcular cambio. \n";
                cout << "7) Consultar historial. \n";
                cout << "8) Finalizar programa. \n";
                cout << "\n";
                cout << "Opcion: ";
                cin >> a;
                cin.ignore();

                switch(a) {

                    case 1:
                    cout << "Ingresa la clave del cliente: ";
                    getline(cin,clave[P][1]);
                    clave[P][1]+= ".txt";
                    archivo.open(clave[P][1].c_str(), ios::in);

                    if(archivo.fail()){
                        cerr << ">>> Nose encentra la clave del cliente " << clave[P][1] << ", la clave del cliente sera registrada al finalizar. \n";
                    }

                    archivo.close();
                    cout << "---------------------------------------------------------------------------------------------------------------------\n";
                    cout << "\n \n";
                    P1[P].setFecha(fecha);
                    cout << "Ingresa el nombres: ";
                    getline(cin,palabras);
                    P1[P].setNombre(palabras);
                    cout << "Ingresa los apellidos: ";
                    getline(cin,palabras);
                    P1[P].setApellido(palabras);

                    try{

                        cout << "Ingresa la cantidad del prestamo: ";
                        cin >> cantidades;
                        P1[P].setCantidadPrestada(cantidades);
                    
                        cout << "Ingresa el porcentaje de interes: ";
                        cin >> cantidades;
                        P1[P].setCantidadDeInteres(cantidades);
                       
                        cout << "Ingresa la cantidad de semanas a pagar: ";
                        cin >> cantidades;
                        P1[P].setCantidadDeSemana(cantidades);

                    }  catch (int){cout << "Alguno de los valores es menor a 0.\n";}


                    acumulador += P1[P].getCantidadPrestada();
                    ganancia += P1[P].Interes();
                    P++;
                    break;

                    case 2:
                    cout << "Ingresa la clave del cliente: ";
                    getline(cin,clave[D][2]);
                    clave[D][2]+= ".txt";
                    archivo.open(clave[D][2].c_str(), ios::in);

                    if(archivo.fail()){
                        cerr << ">>> Nose encentra la clave del cliente " << clave[D][2] << ", la clave del cliente sera registrada al finalizar. \n";
                    }
                    archivo.close();
                    cout << "---------------------------------------------------------------------------------------------------------------------\n";

                    cout << "\n \n";
                    D1[D].setFecha(fecha);
                    cout << "Ingresa el nombres: ";
                    getline(cin,palabras);
                    D1[D].setNombre(palabras);
                    cout << "Ingresa los apellidos: ";
                    getline(cin,palabras);
                    D1[D].setApellido(palabras);
                    try{

                        cout << "Ingresa la cantidad del prestamo: ";
                        cin >> cantidades;
                        D1[D].setCantidadprestada(cantidades);
                    
                        cout << "Ingresa la cantidad devuelta: ";
                        cin >> cantidades;
                        D1[D].setCantidaddevuelta(cantidades);

                        cout << "Ingresa la cantidad de semanas a pagar: ";
                        cin >> cantidades;
                        D1[D].setSemanasapagar(cantidades);
                            
                        cout << "Ingresa el numero de la semana pagada: ";
                        cin >> cantidades;
                        D1[D].setSemanapagada(cantidades);

                    }catch (int) { cout << "Alguno de los valores es menor a 0.\n";}

                    contador += D1[D].getCantidaddevuelta();
                    D++;
                    break;

                    case 3:
                    cout << "Fecha: " << fecha << "\n";
                    cout << "Total de abonos: " << contador << "\n";
                    break;

                    case 4:
                    cout << "Fecha: " << fecha << "\n";
                    cout << "Total de dinero prestado: " << acumulador;
                    break;

                    case 5: 
                    cout << "Fecha: " << fecha << "\n";
                    cout << "Ganancia esperada: " << ganancia;
                    break; 

                    case 6:
                    cout << "Ingresa la cantidad pagada: ";
                    cin >> cantidades;
                    PA1.setPago(cantidades);
                    cout << "Ingresa la cantidad a pagar: ";
                    cin >> cantidades;
                    PA1.setCantidad(cantidades);
                    cout << "Cambio: " << PA1.buelto();

                    break;

                    case 8:
                    cout << "Cerrar programa.";
                    break;

                    case 7:
                    cout << "Ingresa la clave del cliente: ";
                    getline(cin,id);
                    id+= ".txt";
                    archivo.open(id.c_str(), ios::in);
                    cout << "Historial del cliente:\n";
                    cout << "\n";
                    cout << "Fecha de consulta: " << fecha << "\n";
                    string cadena;
                    while(!archivo.eof()){
                        getline(archivo,cadena);
                        cout << cadena << "\n";
                    }
                    archivo.close();
                    break;

                    

                }
            } else { respuesta = 0;}
            i++;
            MAX_ELEMENTS--;
            cout << "\n";
            cout << "Presione 1 se desea repetir y 0 se desea finalizar: ";
            cin >> respuesta;
        } while (respuesta != 0);


        for(int i=0;i<MAX_ELEMENTS; i++){
            
            if (  P1[i].getCantidadPrestada() > 0 && P1[i].getCantidadDeInteres() > 0 && P1[i].getCantidadDeSemana() > 0){
                archivo.open(clave[i][1].c_str(), ios::app);
                archivo << "\n";
                archivo << "Prestamos----------------------------------------------------\n";
                archivo << "Fecha: " << fecha << "\n";
                archivo << P1[i].toString();
                archivo << "-------------------------------------------------------------\n";
                archivo.close();
            }
        }
 
        for(int i=0;i<MAX_ELEMENTS; i++){
            
            if(D1[i].getCantidaddevuelta() > 0 && D1[i].getCantidadprestada() > 0 && D1[i].getSemanapagada() > 0 
                && D1[i].getSemanasapagar() > 0){ ///////
                archivo.open(clave[i][2].c_str(), ios::app);
                archivo << "\n";
                archivo << "Devoluciones-------------------------------------------------\n";
                archivo << "Fecha: " << fecha <<"\n";
                archivo << D1[i].toString();
                archivo << "-------------------------------------------------------------\n";
                archivo.close();
                
                
            }
        }


     


    

    cout << "\n \n";
    cout << "------------------------------------------------------------------------------------------\n";
	cout << "Creador: Jesus A. Mex Damian 24/05/2023.";
    return 0;
}
